### PR TITLE
feat: add redirect function

### PR DESCRIPTION
**Pull request by helintongh**
Design ideas:
when response code is redirect code. 
get Location field and re-request uri which is assigned by location field.
http_redirect

### CODE REVIEW

File: include/cinatra/coro_http_client.hpp [hunk 1]
```diff
@@ -340,6 +354,15 @@ class coro_http_client {
     proxy_bearer_token_auth_token_ = token;
   }
 
+  inline void set_follow_location(const std::string &enable_follow_redirect) {
```
Comment (qicosmos) [inline include/cinatra/coro_http_client.hpp:357]: bool

File: tests/test_cinatra.cpp [hunk 2]
```diff
@@ -350,3 +350,16 @@ TEST_CASE("test coro http bearer token auth request") {
   CHECK(!result.net_err);
   CHECK(result.status == status_type::ok);
 }
+
+TEST_CASE("test coro http redirect request") {
+  coro_http_client client{};
+  std::string uri = "http://httpbin.org/redirect-to?url=http://httpbin.org/get";
+  resp_data result = async_simple::coro::syncAwait(client.async_get(uri));
+  CHECK(!result.net_err);
+  CHECK(client.get_http_status() == "302");
```
Comment (qicosmos) [inline tests/test_cinatra.cpp:359]: status_type --> int

Comment (qicosmos) [inline tests/test_cinatra.cpp:359]: add is_redirect()

File: include/cinatra/coro_http_client.hpp [hunk 3]
```diff
@@ -144,7 +144,17 @@ class coro_http_client {
...
-    return async_request(std::move(uri), http_method::GET, std::move(ctx));
+    resp_data data{};
+    data = co_await async_request(std::move(uri), http_method::GET,
+                                  std::move(ctx));
+    if (enable_follow_redirect_) {
+      if (!redirect_uri_.empty() && is_redirect_resp()) {
+        data = co_await async_request(std::move(redirect_uri_),
+                                      http_method::GET, std::move(ctx));
+        co_return data;
+      }
```
Comment (qicosmos) [inline include/cinatra/coro_http_client.hpp:155]: avoid additional request.

**Comment by helintongh** [comment]
finish it.

File: tests/test_cinatra.cpp [hunk 4]
```diff
@@ -348,18 +348,25 @@ TEST_CASE("test coro http bearer token auth request") {
...
 TEST_CASE("test coro http redirect request") {
   coro_http_client client{};
   std::string uri = "http://httpbin.org/redirect-to?url=http://httpbin.org/get";
   resp_data result = async_simple::coro::syncAwait(client.async_get(uri));
   CHECK(!result.net_err);
-  CHECK(client.get_http_status() == "302");
+  CHECK(result.status == 302);
 
-  client.set_follow_location("on");
+  if (client.is_redirect(result)) {
```
Comment (qicosmos) [inline tests/test_cinatra.cpp:361]: ```if(result.is_redirect())``` is better.

File: tests/test_cinatra.cpp [hunk 5]
```diff
@@ -348,18 +348,25 @@ TEST_CASE("test coro http bearer token auth request") {
...
+    std::string redirect_uri = client.get_redirect_uri();
+    result = async_simple::coro::syncAwait(client.async_get(redirect_uri));
+    CHECK(!result.net_err);
+    CHECK(result.status == 200);
+  }
+
+  client.set_follow_location(true);
```
Comment (qicosmos) [inline tests/test_cinatra.cpp:368]: maybe  the name ```enable_auto_location``` is better than ```set_follow_location```.

File: include/cinatra/coro_http_client.hpp [hunk 6]
```diff
@@ -476,6 +505,13 @@ class coro_http_client {
                                 size_t header_size) {
     // parse header
     const char *data_ptr = asio::buffer_cast<const char *>(read_buf_.data());
+
+    char status[4] = {0};
+    if (data_ptr != nullptr) {
+      memcpy(status, data_ptr + 9, 3);
+      data.status = std::stoi(status);
+    }
```
Comment (qicosmos) [inline include/cinatra/coro_http_client.hpp:513]: get status from parser after http parsing.
data.status = parser.status();

Comment (helintongh) [reply include/cinatra/coro_http_client.hpp]: already modified.

**Pull request merged**